src/Tmds.Ssh/SftpChannel.cs

```diff
@@ -1079,27 +1085,49 @@ private async ValueTask DownloadFileAsync(string remotePath, string localPath, l
             permissions = attributes.Permissions;
         }
 
-        using FileStream localFile = OpenFileStream(localPath, overwrite ? FileMode.Create : FileMode.CreateNew, FileAccess.Write, FileShare.None, permissions!.Value);
+        Stream? dstStream = null;
+        if (destination is string localPath)
+        {
+            dstStream = OpenFileStream(localPath, overwrite ? FileMode.Create : FileMode.CreateNew, FileAccess.Write, FileShare.None, permissions.Value);
+        }
+        else if (destination is Stream stream)
+        {
+            dstStream = stream;
+        }
 
-        ValueTask previous = default;
+        Debug.Assert(dstStream is not null);
 
+        ValueTask previous = default;
         CancellationTokenSource? breakLoop = length > 0 ? new() : null;
 
-        int maxPayload = MaxReadPayload;
-        for (long offset = 0; offset < length; offset += maxPayload)
+        Debug.Assert(breakLoop is not null);
```

Comment:
Can you move this back? `breakLoop` can be null here (when `length` is zero).

Reply:
yes, but i really don't understand the point to be in the loop since everything is assigned before the loop and nothing is reassigned in the loop

Reply:
ah yes, my bad